React/Views/RCTNavigator.m

```diff
@@ -332,6 +345,16 @@ - (void)setPaused:(BOOL)paused
   }
 }
 
+- (void)setInteractivePopGestureEnabled:(BOOL)interactivePopGestureEnabled
+{
+  _interactivePopGestureEnabled = interactivePopGestureEnabled;
+ 
+  _navigationController.interactivePopGestureRecognizer.delegate = self;
+  _navigationController.interactivePopGestureRecognizer.enabled = interactivePopGestureEnabled;
+
+  _popGestureState = interactivePopGestureEnabled ? RCTPopGestureStateEnabled : RCTPopGestureStateDisabled;
+}
+
 - (void)dealloc
 {
   _navigationController.delegate = nil;
```

Comment:
You should nil out `_navigationController.interactivePopGestureRecognizer.delegate` here if it's still set to self.